core/src/wsv.rs

```diff
@@ -645,20 +692,21 @@ impl<W: WorldTrait> WorldStateView<W> {
     pub fn world(&self) -> &W {
         &self.world
     }
-}
-
-impl Deref for World {
-    type Target = Self;
-    #[inline]
-    fn deref(&self) -> &Self::Target {
-        self
-    }
-}
 
-impl DerefMut for World {
-    #[inline]
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        self
+    /// Get triggers set and modify it with `f`
+    ///
+    /// Produces trigger event from `f`
+    ///
+    /// # Errors
+    /// Throws up `f` errors
+    pub fn modify_triggers<F>(&self, f: F) -> Result<(), Error>
+    where
+        F: FnOnce(&TriggerSet) -> Result<TriggerEvent, Error>,
+    {
+        let event = f(&self.triggers).map(Into::into)?;
+        let events: SmallVec<[DataEvent; 1]> = SmallVec(smallvec::smallvec![event]);
```

Comment:
```suggestion
        let events: SmallVec<[DataEvent; 8]> = SmallVec(smallvec::smallvec![event]);
```
Not a requirement, but since there's only one per thread per World we can afford to make it larger.

Reply:
Why we need capacity of `8` if there is only one possible event here? We can use `[DataEvent; 1]` cause we only need possibility to iterate here